### ISSUE TITLE
Automatically add "Export to CodePen" to examples supporting it

### ISSUE THREAD

**Issue by ecraig12345** [OfficeDev/office-ui-fabric-react#7653]
Currently, there are three steps to getting an example to work with "Export to CodePen":
1. Remove any constructs from the example which are incompatible with export to codepen (scss styling, references to any other files not included in the main fabric bundle)
2. Add `// @codepen` to the top of the example file
3. In the `.doc.tsx` for the component, manually add an import of the transformed codepen and set the `codepenJS` property of the relevant example

Step 3 is annoying (and easy to forget) and it seems like we should be able to do it automatically.

(On a related note, it would also be nice if we didn't have to manually add an extra import for the raw example code to the `.doc.tsx`.)

**Comment by dzearing** [comment]
I think steps which make it difficult to provide codepen samples should be blocked in the build so that examples by default follow a format that can be exported.

If there are *specific* build steps we can do to make things less frictiony, we should list them here so that the work is clear.

**Comment by ecraig12345** [comment]
> I think steps which make it difficult to provide codepen samples should be blocked in the build so that examples by default follow a format that can be exported.

How would you propose to do that? The best I can think of is adding an additional lint step which only applies to `**/*.Example.tsx` and enabling the [`no-relative-imports` rule](https://github.com/Microsoft/tslint-microsoft-contrib/blob/master/src/noRelativeImportsRule.ts), plus possibly the [`import-blacklist` rule](https://palantir.github.io/tslint/rules/import-blacklist/) with any files which we know aren't included in the fabric bundle.

> If there are *specific* build steps we can do to make things less frictiony, we should list them here so that the work is clear.

Basically what I wanted was a code transform that runs only on `**/*.doc.tsx* and detects when an example file is imported. It would then add the corresponding code import and codepen import (if the file is codepen-able), as well as the props referencing them. But now that I write that down, I wonder if it would be more trouble than it's worth... And might break bundling of the codepen txt files depending on how we do that.

Another idea is that instead of including the example view (and code+codepen strings) in the doc page props, we instead pass the *path* to the example and let DemoPage.tsx handle the loading. But that would probably complicate bundling *and* lose compile-time path validation.

**Comment by ecraig12345** [comment]
@micahgodbolt Since you removed the backlog review tag, does that mean you have an idea of if/when you'd like it to happen? Even though I opened the issue and think it would be nice to have, it could be challenging, and probably other things are higher priority.

**Comment by ecraig12345** [comment]
Yeah I'm not actively working on this, so it should stay in the backlog.

**Comment by ecraig12345** [comment]
Closing this because we're close to having a full example editor instead. :)
